tests/DFE.FindInformationAcademiesTrusts.CypressTests/cypress/e2e/regression/trusts/trust-contact-page.cy.ts

```diff
@@ -1,23 +1,117 @@
 import trustContactsPage from "../../../pages/trusts/trustContactsPage";
+import commonPage from "../../../pages/commonPage";
+
+function generateNameAndEmail() {
+    const randomNumber = Math.floor(Math.random() * 9999);
+    return {
+        name: `Name${randomNumber}`,
+        email: `email${randomNumber}@education.gov.uk`
+    }
+}
 
 describe("Testing the components of the Trust contacts page", () => {
 
-    describe("On a Trust Overview page with data", () => {
+    describe("On a Trust contacts page with data", () => {
+        beforeEach(() => {
+            cy.login()
+        });
+
+        ['/trusts/contacts?uid=5712', '/trusts/contacts?uid=5527'].forEach((url) => {
+            it(`Can change Trust relationship manager contact details ${url}`, () => {
+                const { name, email } = generateNameAndEmail();
+                cy.visit(url, { failOnStatusCode: false })
+                trustContactsPage
+                    .checkTRMFieldsAndDatasource(name, email)
+            });
+        });
+
+        ['/trusts/contacts?uid=5712', '/trusts/contacts?uid=5527'].forEach((url) => {
+            it(`Can change Schools financial support oversight lead contact details ${url}`, () => {
+                const { name, email } = generateNameAndEmail();
+                cy.visit(url, { failOnStatusCode: false })
+                trustContactsPage
+                    .checkSFSOFieldsAndDatasource(name, email)
+            });
+        });
+
+        ['/trusts/contacts?uid=5712', '/trusts/contacts?uid=5527'].forEach((url) => {
+            it(`Checks a trusts external contact details ${url}`, () => {
+                cy.visit(url, { failOnStatusCode: false })
+                trustContactsPage
+                    .checkAccountingOfficerPresent()
+                    .checkChairOfTrusteesPresent()
+                    .checkChiefFinancialOfficerPresent()
+            });
+        });
```

Comment:
Its a really small one but this can be refactored even further by putting it inside one loop

```suggestion
        ['/trusts/contacts?uid=5712', '/trusts/contacts?uid=5527'].forEach((url) => {
            it(`Can change Trust relationship manager contact details ${url}`, () => {
                const { name, email } = generateNameAndEmail();
                cy.visit(url, { failOnStatusCode: false })
                trustContactsPage
                    .checkTRMFieldsAndDatasource(name, email)
            });
            
            it(`Can change Schools financial support oversight lead contact details ${url}`, () => {
                const { name, email } = generateNameAndEmail();
                cy.visit(url, { failOnStatusCode: false })
                trustContactsPage
                    .checkSFSOFieldsAndDatasource(name, email)
            });
            
            it(`Checks a trusts external contact details ${url}`, () => {
                cy.visit(url, { failOnStatusCode: false })
                trustContactsPage
                    .checkAccountingOfficerPresent()
                    .checkChairOfTrusteesPresent()
                    .checkChiefFinancialOfficerPresent()
            });
        });
```

Reply:
Now done